### PR TITLE
ADD: [droid] Allow suspend on rooted devices

### PR DESCRIPTION
**Pull request by elmerohueso**

### CODE REVIEW

**Comment by koying** [comment]
Just one thing:
Do you confirm the re-enabling of cec is ok?
Do you know if it happens before suspend or after resume?

**Comment by elmerohueso** [comment]
While I don't have any CEC devices myself, I can confirm that the CEC file itself does get marked as re-enabled right after the suspend call (I SFTPd into my box while it was still suspended, and verified that cec = 1 had already been performed).
Also, I haven't gotten any complaints from other users using my addon that makes the same call.

**Comment by koying** [comment]
Cool.

File: xbmc/powermanagement/android/AndroidPowerSyscall.cpp [hunk 1]
```diff
@@ -52,6 +58,21 @@ bool CAndroidPowerSyscall::Powerdown()
   return (rc == 0);
 }
 
+bool CAndroidPowerSyscall::Suspend()
+{
+  if (!m_isRooted)
+    return false;
+
+  if (m_hasCEC)
+    int rc = system((m_su_path + " -c \"echo 0 > " + m_cec_path + "\";" + m_su_path + " -c \"input keyevent KEYCODE_POWER\";" + m_su_path + " -c \"echo 1 > "  + m_cec_path + "\"").c_str());
```
Comment (koying) [inline xbmc/powermanagement/android/AndroidPowerSyscall.cpp:67]: Mm. Doesn't compile. You forgot the brackets.

Comment (elmerohueso) [inline xbmc/powermanagement/android/AndroidPowerSyscall.cpp:67]: Sorry.  I was in a rush (attempting) fixing my fork yesterday.  Which brackets did I miss?  Do you need me to enter them myself, or can you add them on you end and squash it?

Comment (elmerohueso) [inline xbmc/powermanagement/android/AndroidPowerSyscall.cpp:67]: I see it.  Forgot open bracket on line 66 and close on 68.  Sorry.

Comment (koying) [inline xbmc/powermanagement/android/AndroidPowerSyscall.cpp:67]: I'll do the fixup

Comment (elmerohueso) [inline xbmc/powermanagement/android/AndroidPowerSyscall.cpp:67]: Sweet.  I fixed it on mine already, and just won't do another pull request.
